src/router/bundle/bundleGenerator.ts

```diff
@@ -101,7 +102,15 @@ export default class BundleGenerator {
 
         const entries: any = [];
         bundleEntryResponses.forEach(bundleEntryResponse => {
-            const status = bundleEntryResponse.operation === 'create' ? '201 Created' : '200 OK';
+            let status = '200 OK';
+            if (bundleEntryResponse.operation === 'create') {
+                status = '201 Created';
+            } else if (
+                ['read', 'vread'].includes(bundleEntryResponse.operation) &&
+                isEmpty(bundleEntryResponse.resource)

```

Comment:
Will `bundleEntryResponse.resource` be there for the scenario for reading a Patient that isn't there?

Reply:
That case can not happen because an error is thrown in the `persistence` layer when it tries to read a resource that does not exist. We would throw a `ResourceNotFoundError` which would eventually propagate to the ErrorHandler in the router package. The whole Bundle transaction would fail and any changes made would be rolled back.